effekt/shared/src/main/scala/effekt/Lexer.scala

```diff
@@ -489,6 +492,42 @@ class Lexer(source: Source) {
     TokenKind.Str(stringContent.mkString, delim.isMultiline)
   }
 
+  /** Matches a hole containing natural language text.
+   * Holes may contain unquotes, e.g., "${ do foo }" that include arbitrary expressions.
+   * Unlike regular string literals, holes cannot contain escape sequences.
+   * They do emit [[TokenKind.Str]] tokens with the `multiline` flag set to `true`.
+   */
+  def matchHole(continued: Boolean = false): TokenKind = {
```

Comment:
I would have hoped that we could reuse the existing template infrastructure. But eventually we probably want to customize it anyways.

Reply:
I considered this, but it made the template string implementation quite complex by adding new cases. Also there are some differences such as holes not needing escape characters. Therefore I think this is the better trade-off for now.